### PR TITLE
docs: Add logfmt promtail parsing stage

### LINKED ISSUE #5384: logfmt pipeline stage isn't documented
Apparently there is a `logfmt` stage [available](https://github.com/grafana/loki/blob/main/clients/pkg/logentry/stages/logfmt.go) in the promtail pipeline config, but the stage [isn't documented](https://grafana.com/docs/loki/latest/clients/promtail/stages/logfmt/).

### PR DESCRIPTION
**Pull request by alanmcg**
Fixes #5384

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (kavirajk) [inline docs/sources/clients/promtail/stages/logfmt.md:44]:
```suggestion
- `module`: `kafka.consumer.ConsumerFetcherManager`
- `id`: `ConsumerFetcherManager-1382721708341`
```

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (kavirajk) [inline docs/sources/clients/promtail/stages/logfmt.md:18]: nit: instead of referencing `go-kit/log` would be more appropriate to refer the original logfmt blog?
https://brandur.org/logfmt

**Review by kavirajk — COMMENTED** [review]
LGTM! Thanks @alanmcg. Left one minor suggestion.

Leave it to @KMiller-Grafana for final approval.

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/stages/logfmt.md:22]:
```suggestion
## Example
```

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/stages/logfmt.md:44]: Instead of @kavirajk 's suggestion, please use mine. It incorporate's Kavi's as well, but uses the same ordering as the labels appear in the log line source.

```suggestion
The following key-value pairs will be created in the set of extracted data:

- `level`: `info`
- `tag`: `stopping_fetchers`
- `id`: `ConsumerFetcherManager-1382721708341`
- `module`: `kafka.consumer.ConsumerFetcherManager`
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (alanmcg) [reply docs/sources/clients/promtail/stages/logfmt.md:18]: thanks, but I felt it more appropriate to refer to the implementation used, in case of any future bugs this would be a better reference imo

Comment (alanmcg) [reply docs/sources/clients/promtail/stages/logfmt.md:18]: You are right tho, I added an additional link to the brandur.org site

**Review by KMiller-Grafana — APPROVED** [review]

**Comment by alanmcg** [comment]
> 

Thanks, I would love to, been scratching my head trying to understand the cause of the failure but i'm afraid this system is not something I am familiar with

**Comment by cstyan** [comment]
@alanmcg if you rebase your branch off upstream main (or merge it into your branch) it should fix the CI failure

**Comment by alanmcg** [comment]
> @alanmcg can you take care of failing CI? I can merge after that

Thanks, I have done that now. Please merge if everything is as it should be

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (sandeepsukhani) [inline docs/sources/clients/promtail/stages/logfmt.md:42]: `logfmt` stage only extracts the keys that are specified under `mapping`. Since `id` is not specified, it would not be in the extracted data.

File: docs/sources/clients/promtail/stages/logfmt.md
Comment (sandeepsukhani) [inline docs/sources/clients/promtail/stages/logfmt.md:29]: the value part can be empty if you do not want to have data extracted with a different name i.e you can have it configured as below:
```
      level:
      tag:
      module:
```

I think it would be a good idea to club both scenarios in the example like
```
      level:
      tag:
      source: module
```
This would extract the following data:
```
- `level`: `info`
- `tag`: `stopping_fetchers`
- `source`: `kafka.consumer.ConsumerFetcherManager`
```

**Review by sandeepsukhani — CHANGES_REQUESTED** [review]
Added some minor suggestions but other than that it LGTM

**Comment by MichelHollands** [comment]
@alanmcg Hello, can you rebase this and address Sandeep's comments? I'll merge it once that is done.

**Comment by alanmcg** [comment]
> @alanmcg Hello, can you rebase this and address Sandeep's comments? I'll merge it once that is done. 

Thanks, i saw that a similar pull request that covers the missing docs has recently been merged, so i guess it is no longer required

**Comment by MichelHollands** [comment]
Closing as no longer needed

**Pull request closed without merging**
